src/modules/m_restrictchans.cpp

```diff
@@ -42,26 +67,24 @@ class ModuleRestrictChans : public Module
 			newallows.insert(name);
 		}
 		allowchans.swap(newallows);
+
+		// Global config
+		ConfigTag* tag = ServerInstance->Config->ConfValue("restrictchans");
+		allowregistered = tag->getBool("allowregistered", false);
 	}
 
 	ModResult OnUserPreJoin(LocalUser* user, Channel* chan, const std::string& cname, std::string& privs, const std::string& keygiven) CXX11_OVERRIDE
 	{
 		// channel does not yet exist (record is null, about to be created IF we were to allow it)
-		if (!chan)
-		{
-			// user is not an oper and its not in the allow list
-			if ((!user->IsOper()) && (allowchans.find(cname) == allowchans.end()))
-			{
-				user->WriteNumeric(ERR_BANNEDFROMCHAN, cname, "Only IRC operators may create new channels");
-				return MOD_RES_DENY;
-			}
-		}
+		if (!chan && !CanCreateChannel(user, cname))
+			return MOD_RES_DENY;
+
 		return MOD_RES_PASSTHRU;
 	}
 
 	Version GetVersion() CXX11_OVERRIDE
 	{
-		return Version("Only opers may create new channels if this module is loaded",VF_VENDOR);
+		return Version("Allows restricting who can create channels on the network", VF_VENDOR);
```

Comment:
The module is local to the server so this would be more accurate.

```suggestion
		return Version("Allows restricting who can create channels", VF_VENDOR);
```